Source/Core/Resource.js

```diff
@@ -0,0 +1,534 @@
+define([
+    './appendForwardSlash',
+    './Check',
+    './clone',
+    './combine',
+    './defaultValue',
+    './defined',
+    './defineProperties',
+    './freezeObject',
+    './getAbsoluteUri',
+    './getBaseUri',
+    './getExtensionFromUri',
+    './isDataUri',
+    './objectToQuery',
+    './queryToObject',
+    '../ThirdParty/Uri',
+    '../ThirdParty/when'
+], function(appendForwardSlash,
+            Check,
+            clone,
+            combine,
+            defaultValue,
+            defined,
+            defineProperties,
+            freezeObject,
+            getAbsoluteUri,
+            getBaseUri,
+            getExtensionFromUri,
+            isDataUri,
+            objectToQuery,
+            queryToObject,
+            Uri,
+            when) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function parseQuery(queryString) {
+        if (queryString.length === 0) {
+            return {};
+        }
+
+        // Special case we run into where the querystring is just a string, not key/value pairs
+        if (queryString.indexOf('=') === -1) {
```

Comment:
Should this also check for `;` since we learned it is a valid separator as well?

Reply:
No. `;` and `&` are equivalent. This is a check to see if we don't have a key/value query string. eg. Google Earth tile requests are something like this `http://server.com/3d/flatfile?q2-001-q.2`